crates/net/network/src/transactions/fetcher.rs

```diff
@@ -90,10 +90,10 @@ impl TransactionFetcher {
     fn decrement_inflight_request_count_for(&mut self, peer_id: &PeerId) {
         let remove = || -> bool {
             if let Some(inflight_count) = self.active_peers.get(peer_id) {
-                if *inflight_count <= DEFAULT_MAX_COUNT_CONCURRENT_REQUESTS_PER_PEER {
+                *inflight_count -= 1;
```

Comment:
```suggestion
                *inflight_count.saturating_sub(1);
```

better safe than sorry

Reply:
i'm a proponent of using checked methods in case of unintended behavior such as this, thoughts?